### PR TITLE
Connectors

### PR DESCRIPTION
**Pull request by lennyburdette**
---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [x] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [x] Unit Tests
    - [x] Integration Tests
    - [x] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

File: apollo-federation/src/subgraph/mod.rs [hunk 1]
```diff
@@ -327,6 +328,16 @@ impl std::fmt::Debug for ValidSubgraph {
...
+impl From<ValidFederationSubgraph> for ValidSubgraph {
+    fn from(value: ValidFederationSubgraph) -> Self {
+        Self {
+            name: value.name,
+            url: value.url,
+            schema: value.schema.schema().clone(),
+        }
+    }
+}
+
```
Comment (lrlna) [inline apollo-federation/src/subgraph/mod.rs:340]: this can land in `dev` too

File: apollo-federation/src/link/mod.rs [hunk 2]
```diff
@@ -329,6 +331,24 @@ impl Link {
...
+            .iter()
+            .find_map(|directive| {
+                let link = Link::from_directive_application(directive).ok()?;
+                if link.url.identity == *identity {
+                    Some((link, directive))
+                } else {
+                    None
+                }
+            })
+    }
```
Comment (lrlna) [inline apollo-federation/src/link/mod.rs:351]: this can probably also go into `dev`

**Review by lrlna — COMMENTED** [review]
saw two things that can likely land in `dev` right away too

**Comment by abernix** [comment]
> saw two things that can likely land in `dev` right away too

This is really good to call out – do we inventory this somewhere?

**Review by abernix — APPROVED** [review]

**Review by Geal — COMMENTED** [review]
I have only reviewed the parts where the connectors interact with existing router functionality, I'll dig into the connectors plugin itself later.
It's good overall, there's a few things that will need attention, but they're not blockers

**Comment by lennyburdette** [comment]
> saw two things that can likely land in `dev` right away too

done in https://github.com/apollographql/router/pull/5720

File: Cargo.lock [hunk 3]
```diff
@@ -87,7 +87,7 @@ dependencies = [
  "flate2",
  "futures-core",
  "h2",
- "http 0.2.12",
+ "http 0.2.11",
```
[the remark this answers is not in the document]
Comment (lennyburdette) [reply Cargo.lock]: done, excellent suggestion!

File: apollo-router/src/query_planner/subscription.rs [hunk 4]
```diff
@@ -249,10 +249,11 @@ impl SubscriptionNode {
             .and_connection_closed_signal(parameters.subscription_handle.as_ref().map(|s| s.closed_signal.resubscribe()))
             .build();
 
+        // TODO[igni]: refactor so it uses fetchservice
```
[the remark this answers is not in the document]
Comment (lennyburdette) [reply apollo-router/src/query_planner/subscription.rs:252]: i've filed issues for all comments on this PR, thank you!

**Review by dylan-apollo — APPROVED** [review]
I'm ready when you are :shipit:

**Pull request merged**